docker/build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash -ex
+
+export DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC
+apt-get update -y
+apt-get install -y --no-install-recommends \
+    tzdata wget curl openssh-server ssh sudo git-core libibverbs1 ibverbs-providers ibverbs-utils librdmacm1 libibverbs-dev rdma-core libmlx5-1 libssl-dev pkg-config
+
+if [[ ${PYTHON_VERSION} != "3.10" ]]; then
+    apt-get install -y --no-install-recommends software-properties-common
+    add-apt-repository -y ppa:deadsnakes/ppa
+    apt-get update -y
+fi
+
+apt-get install -y --no-install-recommends \
+    rapidjson-dev libgoogle-glog-dev gdb python${PYTHON_VERSION} python${PYTHON_VERSION}-dev python${PYTHON_VERSION}-venv
+apt-get clean -y
+rm -rf /var/lib/apt/lists/*
+
+pushd /opt >/dev/null
+    python${PYTHON_VERSION} -m venv py3
+popd >/dev/null
+
+export PATH=/opt/py3/bin:$PATH
+
+if [[ "${CUDA_VERSION_SHORT}" = "cu128" ]]; then
+    NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_90,code=sm_90 -gencode=arch=compute_120,code=sm_120 -gencode=arch=compute_120,code=compute_120"
+else
+    NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_90,code=sm_90 -gencode=arch=compute_90,code=compute_90"
+fi
+
+pushd /tmp >/dev/null
+    git clone --depth=1 --branch ${NCCL_BRANCH} https://github.com/NVIDIA/nccl.git
```

Comment:
The base image already includes nccl:
```shell
/usr/include/nccl.h
/usr/lib/x86_64-linux-gnu/libnccl.so
```
Is it necessary to build it from source?

Reply:
I think we can build turbomind with a pre-built NCCL wheel, but we need to modify the cmake file to support it. I can have a try to do it.

Reply:
done